useful-scripts/get_data_block_responses/get_data_block_responses.py

```diff
@@ -105,72 +120,63 @@ def print_to_console(text):
-    }
+    def write_this_block(status_code, response_time, response_dict):
+        write_block_to_file(
+            block_id=blockId,
+            release_id=releaseId,
+            subject_id=subjectId,
+            request_dict=queryDict,
+            status_code=status_code,
+            response_time=response_time,
+            response_dict=response_dict,
+        )
 
     block_time_start = time.perf_counter()
     try:
         resp = requests.post(url=url,
-                             headers=headers,
-                             data=query,
-                             timeout=120)
+                             headers={'Content-Type': 'application/json'},
+                             data=json.dumps(queryDict),
+                             timeout=60)
+    except requests.Timeout as e:
+        print_to_console(f'request timeout with block {blockId} subject {subjectId}, {e}')
+        write_this_block(
+            status_code=-1,
+            response_time=-1,
+            response_dict={'error': f'request timeout, {e}'})
+        continue
     except Exception as e:
-        print_to_console(f'request exception with block {guid} subject {subjectId}\nException: {e}')
-        jsonResponse = {'error': f'request exception thrown, {e}'}
+        print_to_console(f'request exception with block {blockId} subject {subjectId}, {e}')
+        write_this_block(
+            status_code=-1,
+            response_time=-1,
+            response_dict={'error': f'request exception thrown, {e}'})
+        continue
     block_time_end = time.perf_counter()
 
-    if resp.status_code != 200:
-        print_to_console(
-            f'Response status wasn\'t 200 for block {guid} '
-            f'subject {subjectId}'
-            f'{resp.text}'
-        )
-
     try:
         jsonResponse = json.loads(resp.text)
     except Exception as e:
-        print_to_console(f'json.loads(resp.text) failed with block {guid} '
-                         f'subject {subjectId}\n'
-                         f'Exception: {e}')
-        jsonResponse = {'error': f'Failed to process response text, Exception: {e}'}
-
-    try:
-        with open(f'{results_dir}/block_{guid}', 'w') as file:
-            file.write(
-                f'block: {guid}\n'
-                f'release: {releaseId}\n'
-                f'subject: {subjectId}\n'
-                f'query:\n'
-                f'{query}\n'
-                f'response status: {resp.status_code}\n'
-                f'time for response: {block_time_end - block_time_start}\n'
-                f'response:\n{json.dumps(jsonResponse, sort_keys=True, indent=2)}'
-            )
-        print_to_console(f'Successfully processed block {guid} for subject {subjectId}!')
-    except BaseException:
-        print_to_console(f'file.write failed with block {guid} '
-                         f'subject {subjectId}\n{resp.text}')
+        print_to_console(f'json.loads(resp.text) failed with block {blockId} subject {subjectId}, {e}')
```

Comment:
Just wondering if `json.loads(resp.text)` is meant to be wrapped in curly brackets (not sure if you're trying to interpolate a value here)?

Reply:
This is intended - I've changed it to text to be clearer.